### DISCUSSION TITLE
drag n drop functionality for combining tabs

### DISCUSSION THREAD

**Discussion by moddypls** [Chatterino/chatterino2#3304] [Ideas]
it would be practical to be able to drag tabs onto one another to combine them. To avoid collision with tab repositioning the merged tab could be previewed in the tab bar while holding the tab or there could be a cursor for repositioning like in firefox when inserting a link into its tab bar (as shown in the image)
![image](https://user-images.githubusercontent.com/71154293/138287968-768805b1-4d37-4484-9ec5-e462cc2ff97f.png)

**Comment by jupjohn** [comment]
Related to #1484 I assume. That issue was closed due to the underlying reason being solved by the tab hiding feature.

Have seen a bit of discussion in chats around this so maybe it's worth having a discussion about. (cc @greencomfytea, you mentioned this I believe)

**Reply by GreenComfyTea** [reply]
Yes, I did. I am not sure that I understood the topic starter's idea correctly thou.
![138287968-768805b1-4d37-4484-9ec5-e462cc2ff97f](https://user-images.githubusercontent.com/30152047/138499585-69f1f193-7a4d-4fea-8eef-25b2049290ce.png)
This icon appears in firefox when dragging a tab from one window to another, right? 
Did he meant it like this: 1st tab + 2nd tab results in one tab with 2 splits (or more, depending on how much splits were in original tabs)? 

Thou It seems to be related to what I want so I gonna describe my ideas.

I imagine tab combining like folders in firefox (or chrome) (see image). 
![image](https://user-images.githubusercontent.com/30152047/138500172-2ec2c09b-db8e-438a-93ec-ae2c6648a998.png)

Why I think we need it: 
1. Helps organizing tabs where they are a lot of them.
2. Lots of tabs take a lot of space, leaving less space for the actual chat.
3. "Toggle visibility of all tabs" is not a solution for some cases. The simplest example is when there are channels where I am modded, and I want to see the tabs all the time to immediately go and read new messages. Personally I find little scenarios when I want to ignore all tabs (and toggle their visibility) and focus on 1 chat, more often I have chat open for the channel i am currently watching, and from time to time I switch to my own channel's tab, channel tabs where I am modded, channels tabs of my friends to read new messages, engage in conversation, etc.
4. Splits are not a solution for some cases. For example, laptops, high dpi, small size screens, single monitor setups or/and low resolution.

What else can help? Smaller height of the tabs (or even adjustable height thru settings). I personally find the height too big. But that's off topic.
